### PR TITLE
Add onboarding configuration wizard to tutorial serverless

### PR DESCRIPTION
**Pull request by digisomni**
image

Give it a go, see how she works and if anything weird/unexpected happens.

If a browser is not supported (such as being on Android) it will give the good ol' QML dialog box asking you for a name.

The main benefit to this is that I can add in literally any configuration options/tutorial info/anything we want to ask/tell/set at a later date.... That'll be very important in tying down configurations and system requirements back to where they should be (much lower than where they are now!)

**Items needed for code review:** defaultscripts.js, away.js, onFirstRun.js, and onEscape.js.

### CODE REVIEW

**Comment by digisomni** [comment]
Oh, and I modified the away script to trigger from a 3rd party script instead of itself, that way the "ESC" functionality can be expanded later.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd.tar_4d701156.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd.tar_4d701156.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd_4d701156.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd_4d701156.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd_4d701156.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd_4d701156.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Quest-Alpha-PR999-57ab4dd_4d701156.apk

**Comment by SilverfishVR** [comment]
Testing [PR999-57ab4dd](https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Alpha-PR999-57ab4dd_4d701156.exe)

It does not trigger for me by default, the only way I could make it run was to manually load `onFirstRun.js ` and then set `"firstRun": true` in interface.json

* position is wack, it's floating off the platform:
Screenshot 2021-02-18 22 31 20

* the defaults are lowest tier possible, especially the forced low framerate makes for a very bad experience if you just click through it.

* the "performance" tab should make clear what it does by listing the target framerates.

* not all tabs/pages are the same size, OCD trigger 😄 

* instead of a one-shot run, it could be a permanent panel in the serverless with a "rerun" button at the end.

* in the future, stuff like audio setup, avatar selection etc. could be added.

**Comment by digisomni** [comment]
All done, it's ready to test again! :)

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f_8d8c141d.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f.tar_8d8c141d.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f.tar_8d8c141d.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f_8d8c141d.sh

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f_8d8c141d.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Quest-Alpha-PR999-08d015f_8d8c141d.apk

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f_8d8c141d.exe

**Comment by SilverfishVR** [comment]
Testing [PR999-08d015f](https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-08d015f_8d8c141d.exe)

looks good, only issue now is that is spawns right in front of the controls guide, we should update the serverless to move that one (and redo it in appropriate style)

**Comment by Aitolda** [comment]
Issue:
In VR mode the Performance card means nothing as your framerate will always be set to real-time.  My first thought was just have it say "affects desktop only" or something like this, however... if you were in VR when you made this selection, it is ignored anyway and won't even impact desktop if you switch back.  Not really sure what the solution is atm.

**Comment by digisomni** [comment]
Is it actually the case that setting your refresh rate profile while in VR will not be counted when switching back to desktop? That sounds a bit like a bug... it should be setting the setting no matter what, no?

**Comment by Aitolda** [comment]
> Is it actually the case that setting your refresh rate profile while in VR will not be counted when switching back to desktop? That sounds a bit like a bug... it should be setting the setting no matter what, no?

Back on stable if I change the refresh rate while in VR it definitely impacts desktop when I switch back.

**Comment by digisomni** [comment]
And doing that in this PR doesn't work...? Even if you use the tablet graphics settings?

**Comment by Aitolda** [comment]
> And doing that in this PR doesn't work...? Even if you use the tablet graphics settings?

Works in the PR too, but not through the tutorial. Only in the graphics tab.

File: interface/resources/serverless/tutorial.json [hunk 1]
```diff
@@ -4,6 +4,56 @@
     "/": "/0.155245,-0.941538,23.9289/0,0.791589,0,0.611053"
 },
   "Entities": [
+      {
+      "id": "{0a199807-4a83-4286-b09c-f21124627c3e}",
```
Comment (ctrlaltdavid) [inline interface/resources/serverless/tutorial.json:8]: Indentation appears to be off.

File: scripts/system/away.js [hunk 2]
```diff
@@ -3,8 +3,6 @@
 //
 //  away.js
 //
-//  examples
-//
 //  Created by Howard Stearns 11/3/15
 //  Copyright 2015 High Fidelity, Inc.
```
Comment (ctrlaltdavid) [inline scripts/system/away.js:7]:
```suggestion
//  Copyright 2015 High Fidelity, Inc.
//  Copyright 2021 Vircadia contributors.
```

File: scripts/system/away.js [hunk 3]
```diff
@@ -329,12 +316,24 @@ var CHANNEL_AWAY_ENABLE = "Hifi-Away-Enable";
 var handleMessage = function(channel, message, sender) {
     if (channel === CHANNEL_AWAY_ENABLE && sender === MyAvatar.sessionUUID) {
         print("away.js | Got message on Hifi-Away-Enable: ", message);
-        setEnabled(message === 'enable');
+        if (message === 'enable') {
+            setEnabled(message === 'enable');
```
Comment (ctrlaltdavid) [inline scripts/system/away.js:320]:
```suggestion
            setEnabled(true);
```

File: scripts/system/onEscape.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+//
+//  onEscape.js
+//
+//  Created by Kalila L. on Feb 3 2021.
+//  Copyright 2020 Vircadia contributors.
```
Comment (ctrlaltdavid) [inline scripts/system/onEscape.js:7]:
```suggestion
//  Copyright 2021 Vircadia contributors.
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-37bb6c3_9b2c6d88.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-37bb6c3.tar_9b2c6d88.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-37bb6c3.tar_9b2c6d88.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-37bb6c3_9b2c6d88.dmg

**Comment by digisomni** [comment]
I tested this with a Valve Index and could not reproduce the issue. Even so, it's quite the edge case so I'm not worried about something like that happening for now. If you have a VR setup then you'll likely know how to double check your graphics settings if things were weird with the Graphics menu.

This wizard is meant to assist the most technically challenged, especially people who are not invested in the platform but are rather here to attend an event or something of the sort.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-121f861-.tar_75dad297.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-121f861-_75dad297.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-121f861-.tar_75dad297.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-121f861-_75dad297.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-121f861-_75dad297.exe

**build (ubuntu-18.04, android)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-PR999-121f861_75dad297.apk
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR999/Vircadia-Quest-Alpha-PR999-121f861_75dad297.apk

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**